library/custom_template/custom_template.php

```diff
@@ -41,210 +40,284 @@ function listitemCode($strDisp, $strInsert)
     }
 }
 
-$contextName = $_REQUEST['contextName'];
-$type = $_REQUEST['type'];
+$contextName = !empty($_GET['contextName']) ? $_GET['contextName'] : 'Plan';
+$type = isset($_GET['type']) ? $_GET['type'] : '';
+$isNN = !empty($type) ? 1 : 0;
 $rowContext = sqlQuery("SELECT * FROM customlists WHERE cl_list_type=2 AND cl_list_item_long=?", array($contextName));
-
 ?>
 <html lang="en">
 <head>
+<?php Header::setupHeader(['common', 'opener', 'jquery-ui', 'select2']); ?>
+<script type="text/javascript" src="<?php echo $GLOBALS['assets_static_relative']; ?>/ckeditor/ckeditor.js"></script>
+<script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/ajax_functions_writer.js"></script>
 
-    <?php Header::setupHeader(['common', 'opener', 'jquery-ui']); ?>
-    <script type="text/javascript" src="<?php echo $GLOBALS['assets_static_relative']; ?>/ckeditor/ckeditor.js"></script>
-    <script type="text/javascript" src="<?php echo $GLOBALS['webroot'] ?>/library/js/ajax_functions_writer.js"></script>
-
-    <script language="JavaScript" type="text/javascript">
-        function refreshme() {
-            top.restoreSession();
-            document.location.reload();
-        }
-        CKEDITOR.config.customConfig = top.webroot_url + '/library/js/nncustom_config.js';
+<script language="JavaScript" type="text/javascript">
+    <?php if (!$isNN) { ?>
         $(function () {
-
-            tabbify();
-
-            $(".iframe_small").on('click', function (e) {
-                e.preventDefault();
-                e.stopPropagation();
-                dlgopen('', '', 330, 120, '', '', {
-                    buttons: [
-                        {text: '<?php echo xla('Close'); ?>', close: true, style: 'default btn-sm'}
-                    ],
-                    onClosed: 'refreshme',
-                    type: 'iframe',
-                    url: $(this).attr('href')
-                });
+            $('#contextSearch').select2({
+                ajax: {
+                    beforeSend: top.restoreSession,
+                    url: top.webroot_url + '/library/ajax/template_context_search.php',
+                    data: function (params) {
+                        let query = {
+                            search: params.term,
+                            type: $(this).attr('id'),
+                            csrf_token_form: "<?php echo attr(collectCsrfToken()); ?>"
```

Comment:
This is pure javascript context, so can use the js escaping function (and drop the outside quotes), which will look like:
```
csrf_token_form: <?php echo js_escape(collectCsrfToken()); ?>
```
(I have 29 more days left :) )